### PR TITLE
INT-3692: Example TCP Annotation Config

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3692

### CODE REVIEW

File: src/reference/asciidoc/ip.adoc [hunk 1]
```diff
@@ -1358,3 +1358,101 @@ The framework includes acknowledgment information in the data packet.
...
+
+[[ip-annotation]]
+=== Annotation-Based Configuration
+
+The following example from the samples respository is used to illustrate some of the configuration options when using
+annotations instead of XML.
+
+[source, java]
+----
+@EnableIntegration <1>
```
Comment (artembilan) [inline src/reference/asciidoc/ip.adoc]: The hint `<1>` looks cool!
Although all other code must be formatted without tabs.
Ins't it ?

File: src/reference/asciidoc/ip.adoc [hunk 2]
```diff
@@ -1358,3 +1358,101 @@ The framework includes acknowledgment information in the data packet.
...
+<3> The entry point to the client-side of the flow. The calling application can `@Autowired` this `Gateway` bean
+and invoke its method.
+
+<4> Outbound endpoints consist of a `MessageHandler` and a consumer that wraps it. In this scenario, the
+`@ServiceActivator` configures the endpoint according to the channel type.
+
+<5> Inbound endpoints (in the TCP/UDP module) are all message-driven so just need to be declared as simple `@Bean` s.
+
+<6> This class provides a number of POJO methods for use in this sample flow (a transformer and service activator
+on the server side, and a transformer on the client side).
```
Comment (artembilan) [inline src/reference/asciidoc/ip.adoc]: I'll change all these EIP mentioning to annotations? Good ?

Comment (garyrussell) [inline src/reference/asciidoc/ip.adoc]: Sounds good; let me know if you want me to fix the tabs or if you will do during merge.

Comment (artembilan) [inline src/reference/asciidoc/ip.adoc]: Yes, I'll do during merge. Just need to know your position.

**Comment by garyrussell** [comment]
BTW, we should make it a background task to add annotation (and DSL) based config examples throughout the docs/samples.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/5cbb2848a108be069e0d4af25d707a9b30081a96

**Pull request closed without merging**